bitmind/download_data.py

```diff
@@ -0,0 +1,93 @@
+from datasets import load_dataset
+import argparse
+import time
+import sys
+import os
+import subprocess
+import glob
+
+from bitmind.constants import DATASET_META, HUGGINGFACE_CACHE_DIR
+
+
+def clear_cache(cache_dir):
+    """Clears lock files and incomplete downloads from the cache directory."""
+    # Find lock and incomplete files
+    lock_files = glob.glob(cache_dir + "/*lock")
+    incomplete_files = glob.glob(cache_dir + "/downloads/**/*.incomplete", recursive=True)
+    try:
+        if lock_files:
+            subprocess.run(["rm", *lock_files], check=True)
+        if incomplete_files:
+            for file in incomplete_files:
+                os.remove(file)
+        print("Hugging Face cache lock files cleared successfully.")
+    except Exception as e:
+        print(f"Failed to clear Hugging Face cache lock files: {e}")
+
+
+def fix_permissions(path):
+    """Attempts to fix permission issues on a given path."""
+    try:
+        subprocess.run(["chmod", "-R", "775", path], check=True)
+        print(f"Fixed permissions for {path}")
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to fix permissions for {path}: {e}")
+
+
+def download_dataset(dataset_path, download_mode: str, cache_dir: str, max_wait: int = 300):
+    """ Downloads the datasets present in datasets.json with exponential backoff
+        download_mode: either 'force_redownload' or 'use_cache_if_exists'
+        cache_dir: huggingface cache directory. ~/.cache/huggingface by default 
+    """
+    retry_wait = 10   # initial wait time in seconds
+    attempts = 0     # initialize attempts counter
+    print(f"Downloading {dataset_path} dataset...")
+    while True:
```

Comment:
issue: Previously attempted downloads may leave incomplete and .lock files. Need to clear cache at the start here, since that is not done in load_dataset. 
@dylanuys